File: criancaController.java
package api;

import application.CriancaService;
import dominio.Crianca;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/criancas")
public class CriancaController {

    @Autowired
    private CriancaService service;

    @PostMapping
    public ResponseEntity<Crianca> criar(@RequestBody Crianca crianca) {
        return ResponseEntity.ok(service.criar(crianca));
    }

    @PutMapping
    public ResponseEntity<Crianca> atualizar(@RequestBody Crianca crianca) {
        return ResponseEntity.ok(service.atualizar(crianca));
    }

    @GetMapping("/{cpf}")
    public ResponseEntity<Crianca> buscarPorCpf(@PathVariable String cpf) {
        return ResponseEntity.ok(service.buscarPorCpf(cpf));
    }

    @GetMapping
    public ResponseEntity<List<Crianca>> buscarTodas() {
        return ResponseEntity.ok(service.buscarTodas());
    }
}

File: criancaController.cs
using application;
using dominio;
using Microsoft.AspNetCore.Mvc;

namespace api;

[ApiController]
[Route("criancas")]
public class CriancaController : ControllerBase
{
    private readonly ICriancaService _service;

    public CriancaController(ICriancaService service)
    {
        _service = service;
    }

    [HttpPost]
    public ActionResult<Crianca> Criar([FromBody] Crianca crianca)
    {
        return Ok(_service.Criar(crianca));
    }

    [HttpPut]
    public ActionResult<Crianca> Atualizar([FromBody] Crianca crianca)
    {
        return Ok(_service.Atualizar(crianca));
    }

    [HttpGet("{cpf}")]
    public ActionResult<Crianca> BuscarPorCpf(string cpf)
    {
        return Ok(_service.BuscarPorCpf(cpf));
    }

    [HttpGet]
    public ActionResult<List<Crianca>> BuscarTodas()
    {
        return Ok(_service.BuscarTodas());
    }
}
